execution/execution-correctness/src/serializer.rs

```diff
@@ -37,15 +40,29 @@ impl SerializerService {
             ExecutionCorrectnessInput::ExecuteBlock(block_with_parent_id) => lcs::to_bytes(
                 &self
                     .internal
-                    .execute_block(block_with_parent_id.0, block_with_parent_id.1),
+                    .execute_block(
+                        id_and_transactions_from_block(&block_with_parent_id.0),
+                        block_with_parent_id.1,
+                    )
+                    .map(|mut result| {
+                        if let Some(prikey) = self.prikey.as_ref() {
+                            let vote_proposal = VoteProposal::new(
+                                result.extension_proof(),
+                                block_with_parent_id.0.clone(),
+                                result.epoch_state().clone(),
+                            );
+                            let signature = prikey.sign_message(&vote_proposal.hash());
+                            result.set_signature(signature);
```

Comment:
The general model should be to encapsulate this rather than amend it.

Specifically, VoteProposal should become SignedVoteProposal or MaybeSignedVoteProposal that way you avoid this nuisance.